From: Benjamin Drung <benjamin.drung@canonical.com>
Date: Sun, 2 Aug 2026 15:31:31 +0200
Subject: fix(net-lib): validate iSCSI LUN parameters

The iscsi module writes the iSCSI LUN into
`/etc/udev/rules.d/99-iscsi-root.rules` and into
`$hookdir/mount/01-$$-iscsi.sh`.

The iSCSI LUN is a number. Validate that given LUNs are numbers. This
prevents invalid characters from propagating to those files.

Origin: upstream, https://github.com/dracut-ng/dracut/commit/05c02db994ca322371a40695a3cc223faa1c7258
---
 modules.d/45net-lib/net-lib.sh | 19 +++++++++++++++++--
 1 file changed, 17 insertions(+), 2 deletions(-)

diff --git a/modules.d/45net-lib/net-lib.sh b/modules.d/45net-lib/net-lib.sh
index 355f0af..0931c1a 100755
--- a/modules.d/45net-lib/net-lib.sh
+++ b/modules.d/45net-lib/net-lib.sh
@@ -365,6 +365,21 @@ normalized_iscsi_name() {
     printf "%s" "$1" | tr '[:upper:]' '[:lower:]' | tr -cd 'a-z0-9.:-'
 }
 
+_validate_iscsi_lun() {
+    local raw_lun="$1"
+
+    case "$raw_lun" in
+        *[!0-9]*)
+            warn "Given iSCSI LUN '$raw_lun' contains non-digits. Stripping non-digit characters."
+            printf "%s" "$raw_lun" | tr -cd '[:digit:]'
+            ;;
+        *)
+            # Purely decimal digits
+            echo "$raw_lun"
+            ;;
+    esac
+}
+
 parse_iscsi_root() {
     local v
     v=${1#iscsi:}
@@ -447,7 +462,7 @@ parse_iscsi_root() {
             iscsi_netdev_name=$1
             shift
         fi
-        iscsi_lun=$1
+        iscsi_lun=$(_validate_iscsi_lun "$1")
         shift
         if [ $# -ne 0 ]; then
             warn "Invalid parameter in iscsi: parameter!"
@@ -465,7 +480,7 @@ parse_iscsi_root() {
         fi
     fi
 
-    iscsi_lun=$1
+    iscsi_lun=$(_validate_iscsi_lun "$1")
     shift
 
     iscsi_target_name=$(printf "%s:" "$@")
